Spanish improvements

amhed opened the pull request:
### PR Checklist (check all)

* [X] Updated `CHANGELOG.md` to describe the included fixes and changes made
* [X] Included issue number on the description of the PR
* [x] Commented on the relevant issue thread about this PR

### Issue number

#187 - Spanish Translations

### Changelog

### Added

* Added a dropdown on account view for choosing language. Displays available langs from resources

### Changed

* Update spanish translation with consistent voice (tu vs usted)
* Changed a few strings that weren't handled by i18n

amhed committed:
187 - Spanish translations

- Minor improvements to spanish dictionary
- Store/Get lang from local storage
- UX dropdown for choosing lang

amhed commented on src/Router.js:
```diff
@@ -14,7 +14,7 @@ import { warningOnline, warningOffline } from './reducers/_warning';
   componentWillMount() {
     lang.init({
-      lng: 'en',
+      lng: 'es',
```
Will revert after rebasing / fix merge conflicts

amhed commented on src/reducers/_account.js:
```diff
@@ -250,6 +250,16 @@ export const accountUpdateAccountAddress = (accountAddress, accountType) => (
 };
 
+export const accountChangeLanguage = language => dispatch => {
+    //TODO: needs to trigger render after change
```
Thinking about best way to trigger a re-render. Either: 
- Wrap the main `Router` component with an HOC using react-i18next (see https://github.com/i18next/react-i18next/blob/master/example/react_withHOC/src/App.js)
- Trigger a forceUpdate?
- Another way I haven't thought about but @pedrouid has?

pedrouid replied:
You are doing great, it's quite simple actually. Because you already control the language selected on the account reducer, you can simply mapStateToProps on the Router and this will be enough to trigger the change.

On the `Router.js`
```
const reduxProps = ({ account }) => ({
  language: account.language
});

export default withRouter(
  connect(reduxProps, {
    warningOffline,
    warningOnline
  })(Router)
);
```

amhed commented:
![image](https://user-images.githubusercontent.com/579291/40274234-8d1645da-5b97-11e8-96a1-120ac3f380a9.png)

Should've pushed this sooner I see 😆

amhed committed:
mapStateToProps

amhed committed:
Changelog

amhed commented:
Rebased + Feedback addressed

pedrouid commented:
>Rebased + Feedback addressed

@amhed nice but the tab menu issue still needs to be addressed

<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

amhed commented:
I have it on another branch. Still struggling with CSS background positioning. Will submit a PR soon-ish (prolly tomorrow)

amhed commented:
Added adaptable TabLayout size and position via: 
- Changing stylized component to receive tabSize prop
- Getting sizes of tab contents via i18n resource loading
- Stop using the single background image for the tab. Now uses a single repeated white image for the center, and `:after` and `:before` pseudo classes to draw the edges
- Added a pixel-per-letter ratio to determine how wide the tab should be

On the last point, would love an opinion no how to improve it, it's a good enough solution, but not perfect. See it working here: http://recordit.co/xEPbXJVDtC

pedrouid commented:
Looking good, definitely needs work on the tab animation, going to take a look soon!

amhed committed:
Using same-length tabs instead. Fixes animations

amhed committed:
Language now loads with option from local storage

Also, fix initial tab size to use correct width

amhed committed:
TabOffset fix

amhed committed:
Finish rebase from upstream/latest

amhed committed:
Simplify .editorconfig

amhed committed:
Rebase once more from latest

amhed commented:
Ended up keeping the same tab size, and only added simple logic to determine first tab offset. I know this will probably change when we add more languages, but at least for ES/EN renders correctly. 

My editor is set to default to 4 spaces. I noticed convention is 2 for the workspace. I added an `.editorconfig` file. It's editor agnostic/most editors have a plugin to enforce.

pedrouid committed:
Merge branch 'latest' into spanish_improvements

ricburton commented:
Tried to pay but there’s a bug in gitcoin.

How much time did you spend on this? It feels like you did a lot of work and we want to reward that.
